### DISCUSSION TITLE
Inquiry about security - input validation

### DISCUSSION THREAD

**Discussion by szilardszabolcs** [grafana/grafana#38536] [Platform]
Hello, I haven't found the information regarding this topic. My company is planning to use Grafana in production environment, but our security team need assurance about some issues.

Does Grafana UI have input validation (as they call it), to prevent malicious code injection?

Thanks in advance!

Best Regards,
Szabolcs

**Comment by ppolewicz** [comment]
Currently even as a viewer you can inject any query to the underlying database, see [this issue](https://github.com/grafana/grafana/issues/18157#issuecomment-972575245) and related for more details

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
